### PR TITLE
feat(kernel): add `initialize_default_services`

### PR DESCRIPTION
**Pull request by hawkw**
One issue with the design of kernel services is that the binding between
services and their clients is not enforced at compile time. This means
that it's possible to add new cross-platform services which should be
spawned on all hardware platforms without the hardware platforms failing
to compile if those services are not spawned.

In order to avoid issues where we forget to spawn a cross-platform
service on a particular platform, such as #164, this branch adds a new
`Kernel::initialize_default_services` that spapwns all the default
cross-platform services and daemon tasks. Rather than having to
duplicate the code to spawn these services in each platform
implementation, we can now simple call this method in the platform impl.
New services which we expect to always be spawned can simply be added
here in the cross-platform kernel, and we don't have to remember to add
code for spawning them to each platform.

To allow the platform to customize settings for the default services,
this method takes a struct containing all the services'
`${SERVICE}Settings` types. In the future, that will probably be
replaced by #160.

While working on this, I also did a bit of cleanup/refactoring of the
various services' registration methods and `Settings` types to make them
a bit more consistent.

Closes https://github.com/tosc-rs/mnemos/issues/164

### CODE REVIEW

**Review by jamesmunns — APPROVED** [review]
I'm okay with merging this, as it's fine *for now*, but I think trying to have one central "all services" list is not something that is super long term viable, and would be much better served as something configuration controlled at compile and/or runtime (e.g. something kconfig+systemctl service list... ish).

We don't HAVE anything like that (and #160 only addresses part of that), so it doesn't make sense to defer something that is better today for something theoretically "better-er" that doesn't exist.

Basically - +1 to merge, but I'd like to get away from this once it is reasonable to.

There are also a lot of dead imports and broken docs links, but feel free to merge once those are resolved.
